pkg/agent/cniserver/pod_configuration.go

```diff
@@ -538,31 +542,36 @@ func (pc *podConfigurator) checkContainerInterface(
 }
 
 func (pc *podConfigurator) checkHostInterface(
-	containerID, vethName string,
+	containerID, podName, podNamespace string,
 	hostIntf, containerIntf *current.Interface,
 	containerLink *vethPair,
 	containerIPs []*current.IPConfig) error {
 	hostVeth, errlink := validateContainerPeerInterface(hostIntf, containerLink)
 	if errlink != nil {
-		klog.Errorf("Failed to check container interface %s peer %s in the host: %v",
-			containerID, vethName, errlink)
+		klog.Errorf("Failed to check container %s interface on the host: %v",
+			containerID, errlink)
 		return errlink
 	}
-	if err := pc.validateOVSPort(hostVeth.name, containerIntf.Mac, containerID, containerIPs); err != nil {
+	if err := pc.validateOVSInterfaceConfig(hostVeth.name,
+		containerIntf.Mac,
+		containerID,
+		podName,
+		podNamespace,
+		containerIPs); err != nil {
 		klog.Errorf("Failed to check host link %s for container %s attaching status on ovs. err: %v",
 			hostVeth.name, containerID, err)
 		return err
 	}
 	return nil
 }
 
-func (pc *podConfigurator) validateOVSPort(
-	ovsPortName, containerMAC, containerID string,
+func (pc *podConfigurator) validateOVSInterfaceConfig(
+	interfaceName, containerMAC, containerID, podName, podNamespace string,
 	ips []*current.IPConfig) error {
-	if containerConfig, found := pc.ifaceStore.GetInterface(ovsPortName); found {
+	if containerConfig, found := pc.ifaceStore.GetContainerInterface(podName, podNamespace); found {
```

Comment:
Just curious, since interfaceName(hostVeth.name) is the same as the expected result of function `GetContainerInterface(podName, podNamespace)`, why regenerate it every time?  If the API is defined with at least two parameters(podName and podNamespace), maybe interfaceName is not needed to provide.

Reply:
But in this PR, I changed the InterfaceStore to use another interface key to be the map's key, no more the interface name (assuming interface name could change across Antrea versions), so we have to re-generate the key string for every InterfaceConfig lookup.

I think here the interfaceName is provided just for appending the information to the returned error, but you are right I could remove it.

But you reminded me one thing: in validatePrevResult() we get hostIntf by comparing the interface name. Do we have another way to get the interface from prevResult.Interfaces? For example, could we assume prevResult.Interfaces should contain only two interfaces, the other one not matching the container interface must be the host interface?

Reply:
I updated the PR that removes the interfaceName argument of this func, and also changes validatePrevResult() and checkHostInterface() logic to look up the host interface by matching the ifIndex and peerIndex. Please take another look.